LSP-lua.sublime-settings

```diff
@@ -159,5 +159,61 @@
+            // [wiki]zh:https://github.com/CppCXY/EmmyLuaCodeStyle
+            // [basic code reformat option]
+            "indent_style": "space",
+            "indent_size": "4",
+            "tab_width": "4",
+            "quote_style": "none",
+            "call_arg_parentheses": "keep",
+            "continuation_indent_size": "4",
+            "local_assign_continuation_align_to_first_expression": "false",
+            "align_call_args": "false",
+            "align_function_define_params": "true",
+            "align_chained_expression_statement": "false",
+            "keep_one_space_between_table_and_bracket": "true",
+            "align_table_field_to_first_field": "false",
+            "keep_one_space_between_namedef_and_attribute": "true",
+            "max_continuous_line_distance": "1",
+            "continuous_assign_statement_align_to_equal_sign": "true",
+            "continuous_assign_table_field_align_to_equal_sign": "true",
+            "label_no_indent": "false",
+            "do_statement_no_indent": "false",
+            "if_condition_no_continuation_indent": "false",
+            "if_branch_comments_after_block_no_indent": "false",
+            "table_append_expression_no_space": "false",
+            "if_condition_align_with_each_other": "false",
+            "long_chain_expression_allow_one_space_after_colon": "false",
+            "remove_empty_header_and_footer_lines_in_function": "true",
+            "remove_expression_list_finish_comma": "true",
+            "end_of_line": "auto",
+            "detect_end_of_line": "false",
+            "keep_line_after_if_statement": "minLine:0",
+            "keep_line_after_do_statement": "minLine:0",
+            "keep_line_after_while_statement": "minLine:0",
+            "keep_line_after_repeat_statement": "minLine:0",
+            "keep_line_after_for_statement": "minLine:0",
+            "keep_line_after_local_or_assign_statement": "keepLine",
+            "keep_line_after_function_define_statement": "keepLine:1",
+            "keep_line_after_expression_statement": "keepLine",
+            // [diagnostic]
+            "enable_check_codestyle": "true",
+            "max_line_length": "120",
+            "insert_final_newline": "true",
+            // [name style check]
+            "enable_name_style_check": "false",
+            "local_name_define_style": "snake_case",
+            "function_param_name_style": "snake_case",
+            "function_name_define_style": "snake_case",
+            "local_function_name_define_style": "snake_case",
+            "table_field_name_define_style": "snake_case",
+            "global_variable_name_define_style": "snake_case|upper_snake_case",
+            "module_name_define_style": "same('m')|same(filename, snake_case)",
+            "require_module_name_style": "same(first_param, snake_case)",
+            "class_name_define_style": "same(filename, snake_case)"
```

Comment:
These seem to be new server settings.

For all of these server settings, there is an (optional) integration with LSP-json so that you can get auto-complete for all the different settings. It works by making LSP-json read and parse this file: https://github.com/sublimelsp/LSP-lua/blob/main/sublime-package.json

You can automatically (re)generate this sublime-package file by copying this URL into your clipboard:

https://raw.githubusercontent.com/sumneko/vscode-lua/v3.3.0/package.json

and then running the command "*LSP Development: Parse VSCode Package JSON From Clipboard*". This will create two read-only buffers in your sublime window that you can compare to the old sublime-package.json. Adjust accordingly.

Reply:
I see now that the package.json from vscode-lua doesn't help:

https://github.com/sumneko/vscode-lua/blob/199785359b25903a6634c17f0597a0aa18c88ea0/package.json#L1142-L1155

Just to get some context for me, as I understand the formatter is a new feature and it uses this backend: https://github.com/CppCXY/EmmyLuaCodeStyle

You can apparently also specify these formatter options in an .editorconfig file, and this EmmyLuaCodeStyle formatter will pick that up: https://github.com/sumneko/lua-language-server/blob/master/.editorconfig

Question: what takes precedence? Do the server settings override the .editorconfig, or does the .editorconfig override the server settings?

If the server settings take precedence over the .editorconfig, then we are silently modifying all of the user's existing formatting options. This would be a bad experience.

If the .editorconfig takes precedence over the server settings, then I'm okay with these changes.

Reply:
By the way, thank you for upgrading to a newer version!